mava/wrappers/pettingzoo.py

```diff
@@ -444,6 +445,11 @@ def step(self, actions: Dict[str, np.ndarray]) -> dm_env.TimeStep:
 
         # Get valid actions for active agents
         actions = {key: actions[key] for key in self.agents}
+
+        # Convert Jax device array actions to python integers
+        if not all(type(value) == int for value in actions.values()):  # type: ignore
```

Comment:
Maybe we should consider doing this in the executor? I had this in a different branch - https://github.com/instadeepai/Mava/commit/19c19e27adb2e4ecfe3da144b469e31e23226888. What do you think? We might need to benchmark this.

Reply:
I considered doing this, but the problem seems to only be affecting PettingZoo. By only changing right before stepping the environment we also still store device arrays for use during training later on? What are your thoughts?